lib/b-tree.js

```diff
@@ -0,0 +1,318 @@
+'use strict';
+
+const DEFAULT_DEGREE = 6; // min degree of b-tree.
+// All vertices except the root have [degree ... 2 * degree] child nodes
+
+function BTree(degree = DEFAULT_DEGREE) {
+  this.root = new BTreeNode();
+  this.minDegree = degree;
+}
+
+function BTreeNode(leaf = true) {
+  this.elements = [];
+  this.leaf = leaf;
+  this.lastChild = null;
+}
+
+function Element(value, typle) {
+  this.value = value;
+  this.typle = typle;
+  this.child = null;
+}
+
+BTree.prototype.getEqual = function(value) {
+  let curNode = this.root;
+  while (true) {
+    let needChangedNodeFlag = true;
+    for (const curElement of curNode.elements) {
+      if (value <= curElement.value) {
+        if (curElement.value === value) return curElement.typle;
+        if (curNode.leaf) return null;
+        curNode = curElement.child;
+        needChangedNodeFlag = false;
+        break;
+      }
+    }
+    if (curNode.leaf) return null;
+    if (needChangedNodeFlag) {
+      curNode = curNode.lastChild;
+    }
+  }
```

Comment:
Whole loop looks strange and could be rewritten to avoid unnecessary checks and flags as well as `break` usage.
Also, it will be better to write separate `find` function to find element equals-or-before given in a list and use it here.